pkg/runctl/controller.go

```diff
@@ -364,7 +368,9 @@ func (c *Controller) syncHandler(key string) error {
 		pipelineRun.StoreErrorAsMessage(err, "failed to load configuration for pipeline runs")
 		c.metrics.CountResult(pipelineRun.GetStatus().Result)
 
-		return c.finish(pipelineRun)
+		if err := c.finish(pipelineRun); err != nil {
```

Comment:
We should return here in any case as before.

Reply:
Yes, true. Don't know why I changed that. In case `c.finish(./.)` succeeds to return nil. Wouldn't it more appropriate to return `err` in that case (`err` as it has been received from `c.loadPipelineRunsConfig()`.